src/lib/eslint/EslintCommons.ts

```diff
@@ -84,31 +91,30 @@ export class EslintProcessHelper {
 	addRuleResultsFromReport(
 		engineName: string,
 		results: RuleResult[],
-		report: ESReport,
-		ruleMap: Map<string, ESRule>,
+		esResults: ESResult[],
+		ruleMap: Map<string,ESRuleMetadata>,
 		processRuleViolation: (fileName: string, ruleViolation: RuleViolation) => void): void {
-			for (const r of report.results) {
-			// Only add report entries that have actual violations to report.
+		esResults.forEach(r => {
 			if (r.messages && r.messages.length > 0) {
 				results.push(this.toRuleResult(engineName, r.filePath, r.messages, ruleMap, processRuleViolation));
 			}
-		}
+		});
 	}
 
 	toRuleResult(
 		engineName: string,
 		fileName: string,
 		messages: ESMessage[],
-		ruleMap: Map<string, ESRule>,
+		ruleMap: Map<string, ESRuleMetadata>,
 		processRuleViolation: (fileName: string, ruleViolation: RuleViolation) => void): RuleResult {
 		return {
 			engine: engineName,
 			fileName,
 			violations: messages.map(
-				(v): RuleViolation => {
-					const rule = ruleMap.get(v.ruleId);
-					const category = rule ? rule.meta.docs.category : "";
-					const url = rule ? rule.meta.docs.url : "";
+				(v: ESMessage): RuleViolation => {
+					const ruleMeta = ruleMap.get(v.ruleId);
+					const category = ruleMeta ? ruleMeta.type : "problem";
```

Comment:
At this point, all rules should have types either because their metadata defines one or because we defaulted to `problem`. But just in case, here we'll also default to `problem`.